orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/EvaluateDeploymentHealthTask.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.monitoreddeploy;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.config.DeploymentMonitorServiceProvider;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthRequest;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthResponse;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.StatusReason;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "monitored-deploy.enabled")
+public class EvaluateDeploymentHealthTask extends MonitoredDeployBaseTask {
+  @Autowired
+  EvaluateDeploymentHealthTask(
+      DeploymentMonitorServiceProvider deploymentMonitorServiceProvider, Registry registry) {
+    super(deploymentMonitorServiceProvider, registry);
+  }
+
+  @Override
+  public TaskResult executeInternal() {
+    EvaluateHealthRequest request = new EvaluateHealthRequest(stage);
+
+    log.info(
+        "Evaluating health of deployment {} at {}% with {}",
+        request.getNewServerGroup(), request.getCurrentProgress(), monitorDefinition);
+
+    EvaluateHealthResponse response = monitorDefinition.getService().evaluateHealth(request);
+    sanitizeAndLogResponse(response);
+
+    EvaluateHealthResponse.NextStepDirective directive = response.getNextStep().getDirective();
+
+    Id statusCounterId =
+        registry
+            .createId("deploymentMonitor.healthStatus")
+            .withTag("monitorId", monitorDefinition.getId())
+            .withTag("status", directive.toString());
+
+    registry.counter(statusCounterId).increment();
+
+    if (directive != EvaluateHealthResponse.NextStepDirective.WAIT) {
```

Comment:
why not time the wait?

Reply:
because it will be implied when the operation actually completes, e.g. when the monitor actually returns `complete` or `abort` we will have the time it needed to reach that decision, which, i think, is the interesting time 

though now that I think about it, I don't take into account the case when the monitor just takes so long that it times out, let me add it to my todos

Reply:
I see, can you maybe add a useful comment in the code so that I don't ask the same question a week from now? :)